src/cpp/src/continuous_batching_pipeline.cpp

```diff
@@ -140,7 +140,14 @@ class ContinuousBatchingPipeline::Impl {
         sampling_params.set_eos_token_id(m_tokenizer.get_eos_token_id());
         sampling_params.validate();
         SequenceGroup::Ptr sequence_group = std::make_shared<SequenceGroup>(request_id, input_ids,
-                                                                            sampling_params, m_scheduler->get_config().block_size);
+                                                                            sampling_params, 
+                                                                            m_scheduler->get_config().block_size,
+                                                                            m_scheduler->get_config().enable_prefix_caching);
+        if (m_scheduler->get_config().enable_prefix_caching) {
+            sequence_group->set_sequence_group_ptr(sequence_group);
```

Comment:
Tried to add
`class SequenceGroup : public std::enable_shared_from_this<SequenceGroup>`
But following line SequenceGroup constructor fails:
`std::weak_ptr<SequenceGroup> gr = shared_from_this();`

Error:
```
unknown file: Failure
C++ exception with description "bad_weak_ptr" thrown in the test body.
```

So looks like it can't be used in class constructor.

Reply:
ok, let's keep as is